src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs

```diff
@@ -187,7 +191,41 @@ private IEnumerable<MongoUpdate> ConvertUpdateEntriesToMongoUpdates(IList<IUpdat
 
     private static MongoUpdate ConvertAddedEntryToMongoUpdate(string collectionName, string id, IUpdateEntry entry)
     {
-        throw new NotImplementedException(); // EF-17
+        var entityType = entry.EntityType;
```

Comment:
I'm still confused about which interfaces we are supposed to use.

I thought we were supposed to use `IReadOnlyEntityType`, but `entry.EntityType` is of type `IEntityType`.

I guess we will continue to use `IReadOnlyEntityType` (instead of `IEntityType`) whenever possible?

Everything still compiles because `IEntitType` implements `IReadOnlyEntityType`.

Reply:
IReadOnlyEntityType is a version of IEntityType without the methods/properties to modify it. Ideally after the DbContext is created everything should be IReadOnlyEntityType but it sounds like they have a few places where this isn't the case. I'd not worry about it but I can mention it on the next call with the EF team.